src/appleseed/renderer/kernel/lighting/bdpt/bdptlightingengine.cpp

```diff
@@ -51,10 +54,45 @@ namespace
     // Bidirectional Path Tracing lighting engine.
     //
 
+    enum class BDPTVertexType
+    {
+        Camera, Light, Surface, Medium
+    };
+
+    struct BDPTVertex
+    {
+        foundation::Vector3d    m_position;
```

Comment:
You can remove all `foundation::` prefixes since there's already a using clause.

I'd also suggest adding a `using namespace std` clause to make the code terser.